modules/oai.xql

```diff
@@ -0,0 +1,135 @@
+xquery version "3.1" encoding "UTF-8";
+
+(:~
+ :  Module for exporting data as oai/xml
+ :  see https://www.openarchives.org/OAI/openarchivesprotocol.html
+ :)
+
+declare namespace tei="http://www.tei-c.org/ns/1.0";
+declare namespace mei="http://www.music-encoding.org/ns/mei";
+declare namespace output="http://www.w3.org/2010/xslt-xquery-serialization";
+declare namespace request="http://exist-db.org/xquery/request";
+declare namespace response="http://exist-db.org/xquery/response";
+declare namespace util="http://exist-db.org/xquery/util";
+declare namespace oai="http://www.openarchives.org/OAI/2.0/";
+
+import module namespace functx="http://www.functx.com";
+import module namespace crud="http://xquery.weber-gesamtausgabe.de/modules/crud" at "crud.xqm";
+import module namespace config="http://xquery.weber-gesamtausgabe.de/modules/config" at "config.xqm";
+import module namespace wega-util="http://xquery.weber-gesamtausgabe.de/modules/wega-util" at "wega-util.xqm";
+import module namespace date="http://xquery.weber-gesamtausgabe.de/modules/date" at "xmldb:exist:///db/apps/WeGA-WebApp-lib/xquery/date.xqm";
+import module namespace str="http://xquery.weber-gesamtausgabe.de/modules/str" at "xmldb:exist:///db/apps/WeGA-WebApp-lib/xquery/str.xqm";
+import module namespace mycache="http://xquery.weber-gesamtausgabe.de/modules/cache" at "xmldb:exist:///db/apps/WeGA-WebApp-lib/xquery/cache.xqm";
+import module namespace lod="http://xquery.weber-gesamtausgabe.de/modules/lod" at "lod.xqm";
+import module namespace lang="http://xquery.weber-gesamtausgabe.de/modules/lang" at "lang.xqm";
+import module namespace query="http://xquery.weber-gesamtausgabe.de/modules/query" at "query.xqm";
+import module namespace wdt="http://xquery.weber-gesamtausgabe.de/modules/wdt" at "wdt.xqm";
+
+declare option output:method "xml";
+declare option output:media-type "application/xml";
+declare option output:indent "yes";
+
+(:~
+ : Get the last date of modification from dataHistory.xml. Fallback: VersionDate from options.xml
+ : @author Dennis Ried 
+ :
+ : @param $docID The ID of the document
+ : return The date ad xs:dateTime or empty 
+:)
+declare %private function oai:last-modified($docID) as xs:dateTime {
+    let $props := config:get-data-props($docID)
```

Comment:
this function is called `config:get-svn-props`. A quick fix would be to adjust it here, the more elaborate way would be to duplicate such a function in the config module as e.g. `config:get-version-control-props`, link the old one to the new function and deprecate the old one, like:
```xquery
(:~
 : Deprecated, please use config:get-version-control-props#1
 : This function is to be removed in the near future
:)

declare function config:get-svn-props($docID as xs:string) as map(*) {
    config:get-version-control-props($docID),
    wega-util:log-to-file('info', 'call to deprecated function `config:get-svn-props`!')
};
```

Reply:
yes, in my code it will be overwritten and the different name was to avoid myself to get confused. I'll fix that!